eqs/src/routes.rs

```diff
@@ -178,6 +180,22 @@ pub async fn get_transaction_by_hash(
     }
 }
 
+///Return an AssetDefinition, making JSON-RPC connection optional
+pub async fn get_contract_address(
+    bindings: &HashMap<String, RouteBinding>,
+    query_result_state: &QueryResultState,
+) -> Result<Option<Address>, tide::Error> {
+    if let Some(addr) = query_result_state
+        .address_from_assetdef
+        .get(&bindings[":assetdef"].value.to::<AssetDefinition>()?)
+        .cloned()
+    {
+        Ok(Some(addr))
+    } else {
+        Ok(None)
+    }
+}
+
```

Comment:
Realizing now that this doesn't check for the wallet asking for a bad/nonexistent AssetDefinition. Could be solved by adding <native AssetDefinition, None> to the address_from_assetdef map- could this be done during the eth_polling new function?

Reply:
Not sure what you mean. If the wallet asks for an `AssetDefinition` that isn't in the mapping, this will return `None`, which is the correct behavior. FWIW though I think you could eliminate the `if` statement and just return `Ok(query_result_state....cloned())`